rust/src/web/rest/blocks.rs

```diff
@@ -30,22 +30,21 @@ pub async fn get_blocks(
 
     if let Ok(Some(best_tip)) = db.get_best_block() {
         let mut best_chain: Box<Vec<PrecomputedBlock>> = Box::new(vec![best_tip.clone()]);
-        let mut counter = 1;
         let mut parent_state_hash = best_tip.previous_state_hash();
 
         loop {
-            if counter == limit {
-                break;
-            }
             if let Ok(Some(block)) = db.get_block(&parent_state_hash) {
                 parent_state_hash = block.previous_state_hash();
                 best_chain.push(block);
             } else {
                 // No parent
                 break;
             }
-            counter += 1;
+            if best_chain.len() == limit as usize {
+                break;
+            }
```

Comment:
Could this ever get stuck and never hit the break?

Reply:
It's semantically equivalent to what was there, just without the `counter` variable. Each time we loop, it either pushes a block or breaks if it has no parent